README.md

```diff
@@ -6,32 +6,31 @@ JavaScript library to validate and transform Cardano transactions accroding to [
+* transaction - transaction as defined in the [shelley-ma.cddl#13](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/shelley-ma/test-suite/cddl-files/shelley-ma.cddl#L13)
 * raw transaction - transaction outputed by `cardano-cli transaction build-raw`
 
-we will refer to them collectively as a *transaction*.
 
 ### `parse`
-Parses a CBOR encoded *transaction* into a Transaction object. If the CBOR is malformed the parser will throw an error. Available methods:
+Parses a CBOR encoded input into a `TransactionBody`, `Transaction` or `RawTransaction`, respectively. If the CBOR is malformed the parser will throw an error. Available methods:
 ```ts
 parseTxBody(txBodyCbor: Buffer) => TransactionBody
-parseSignedTx(signedTxCbor: Buffer) => SignedTransaction
+parseTx(txCbor: Buffer) => Transaction
 parseRawTx(rawTxCbor: Buffer) => RawTransaction
 ```
-When parsing a signed/raw transaction fields other than the transaction body are only decoded from CBOR and are not parsed according to the CDDL, in TypeScript they have type `unknown`.
-
 The `TransactionBody` object mostly follows the CDDL, but sometimes makes small deviations for better DX, such as parsing certain tuples as objects, or maps as arrays because working with maps is cumbersome in JavaScript.
 
+When parsing a transaction or raw transaction, fields other than the transaction body are only decoded from CBOR and are not parsed according to the CDDL, in TypeScript they have type `unknown`.
+
 ### `encode`
-Takes a Transaction object and encodes it back to CBOR using canonical CBOR serialization format as specified in [Section 3.9 of CBOR RFC](https://datatracker.ietf.org/doc/html/rfc7049#section-3.9). Available methods:
+Takes a `TransactionBody`, `Transaction` or `RawTransaction` object and encodes it back to CBOR using canonical CBOR serialization format as specified in [Section 3.9 of CBOR RFC](https://datatracker.ietf.org/doc/html/rfc7049#section-3.9). Available methods:
 ```ts
 encodeTxBody(txBody: TransactionBody) => Buffer
-encodeSignedTx(signedTx: SignedTransaction) => Buffer
+encodeTx(tx: Transaction) => Buffer
 encodeRawTx(rawTx: RawTransaction) => Buffer
 ```
 
 ### `validate`
-Takes a CBOR encoded *transaction*, validates it according to `CIP-0021` and returns an array of found validation errors.
+Takes a CBOR encoded input, validates it according to `CIP-0021` and returns an array of found validation errors.
 The validation errors are of type `ValidationError` and we distinguish between *fixable* and *unfixable* errors. If the `validate` function reports:
 * no *fixable* and no *unfixable* errors then the CBOR is already fully compliant with CIP-0021.
 * only *fixable* errors then a call to a `transform` function will solve the found errors and the *transcation* will be compliant with CIP-0021.
```

Comment:
We should mention that tx length can be increased or decreased which might affect the minimum required fee.
```suggestion
* only *fixable* errors then a call to a `transform` function will solve the found errors and the *transaction* will be compliant with CIP-0021 (note: the length of the resulting CBOR might be increased or decreased which might affect the minimum required fee).
```

Reply:
I added the note to the `transform` paragraph.